### PR TITLE
Interactive dispatch plot with plotly

### PR DESCRIPTION
**Pull request by jnnr**
This adds an interactive dispatch plot. 

TODO:
* [x] Test it
* [x] Map colors correctly
* [x] Set y-axis labels
* [x] Set order of stacking

The timeseries on this branch are not very interesting, therefore you might want to use some from oemof-B3 to test it.

### CODE REVIEW

File: oemoflex/tools/plots.py [hunk 1]
```diff
@@ -153,6 +154,71 @@ def group_agg_by_column(df):
...
+    ----------
+    df : pandas.DataFrame
+        DataFrame with replaced near zeros.
+    """
+    tolerance = 1e-3
+    df[abs(df) < tolerance] = 0.0
+    return df
+
+
+def preprocessing(df, bus_name, demand_name):
```
Comment (jnnr) [inline oemoflex/tools/plots.py:177]: This function needs a more specific name.

File: oemoflex/tools/plots.py [hunk 2]
```diff
@@ -179,6 +246,181 @@ def filter_timeseries(df, start_date=None, end_date=None):
...
+    layout = dict(font=dict(family="Aleo"))
+
+    fig = dict(data=traces, layout=layout)
+
+    fig = go.Figure(fig)
+
+    return fig
+
+
+def plot_dispatch_plotly2(
```
Comment (jnnr) [inline oemoflex/tools/plots.py:338]: If this is the more advanced version of the original function, it is ok to replace.

File: oemoflex/tools/plots.py (hunk 1 above)
Comment (juliusmeier) [reply oemoflex/tools/plots.py:177]: What do you think about "prepare_plotting_data" ? Or rather more specific?

Comment (jnnr) [reply oemoflex/tools/plots.py:177]: prepare_dispatch_data?

Comment (juliusmeier) [reply oemoflex/tools/plots.py:177]: Yes, I like that

**Comment by jnnr** [comment]
Looks great! I was wondering if we can add units to the y-axis as in the static plot, @juliusmeier? Apart from that, this seems ready to be merged.

**Comment by juliusmeier** [comment]
> Looks great! I was wondering if we can add units to the y-axis as in the static plot, @juliusmeier? Apart from that, this seems ready to be merged.

Ah yes, I think we cannot use the eng_formatter but I will look for another solution.

File: oemoflex/tools/plots.py [hunk 3]
```diff
@@ -179,6 +246,124 @@ def filter_timeseries(df, start_date=None, end_date=None):
...
+    else:
+        raise ValueError(
+            key,
+            " has both, negative and positive values. But it should only have either one",
+        )
+
+    return stackgroup
+
+
+def plot_dispatch_plotly(
```
Comment (jnnr) [inline oemoflex/tools/plots.py:281]: Should we allow to pass a custom mapping as well?

Comment (juliusmeier) [reply oemoflex/tools/plots.py:281]: Yes, might be a good idea to make it more accessable.

**Comment by jnnr** [comment]
* [ ] Let eng_format return ax as well
* [ ] Is this necessary and if yes, why does it not happen in plot_dispatch_plotly()?:
    df[df.select_dtypes(include=["number"]).columns] *= conv_number

**Comment by juliusmeier** [comment]
>     Is this necessary and if yes, why does it not happen in plot_dispatch_plotly()?:
>       df[df.select_dtypes(include=["number"]).columns] *= conv_number

I found out that it is not necessary since all columns should always contain numbers. The timedata is in the index column and not affected by the operation ```df *= conv_number```

File: oemoflex/tools/plots.py (hunk 3 above)
Comment (jnnr) [reply oemoflex/tools/plots.py:287]: Will be done in a new PR.

**Pull request merged**